src/test/test-dlopen-so.c

```diff
@@ -75,6 +76,10 @@ static int run(int argc, char **argv) {
         assert_se(dlopen_libarchive() >= 0);
 #endif
 
+#if HAVE_ZSTD
```

Comment:
Wrong macro for lz4. Also, you should expose the dlopen() functions for all the compression libraries and call each of them here.